src/services/UserService.ts

```diff
@@ -1,19 +1,64 @@
-import { UploadedPromotionsResponse } from '../types/api';
+import axios, { AxiosResponse } from 'axios';
+
+import { GetUserResponse, UpdateUserResponse, UploadedPromotionsResponse } from '../types/api';
 import { Promotion } from '../types/promotion';
+import { User } from '../types/user';
 import { isError } from '../utils/api';
 import Routes from '../utils/routes';
 
 class UserService {
-  public async getUploadedPromotions(userId: string): Promise<Promotion[]> {
-    const url = Routes.USERS.UPLOADED_PROMOTIONS(userId);
-    return fetch(url)
-      .then((response: Response) => response.json())
-      .then((response: UploadedPromotionsResponse) => {
-        if (isError<UploadedPromotionsResponse>(response)) {
-          return Promise.reject(response);
+  private userId: string;
+
+  public constructor() {
```

Comment:
Would we save userId as a cookie after the user authenticates, and have this constructor get the userId from the cookie every time the web app is reopened?

Reply:
@yunishimura0716 do you mind answering this question?

Reply:
I would say yes.
I'm now thinking about the best way to store user info (`userId` in this case), Local storage or cookie or other??.

Reply:
I think local storage is probably a good and common solution! We can persist it to local storage for a specific amount of time and then when it's invalidated, just log out the user

Reply:
@richardyjtian I realized that it is impossible to get `userId` for current flow of `signIn` since it just authorize by firebase server, not our db. I will change the implementation of BE to be able to request GET `users/:id` where `id` is user id of firebase not our db. So, we don't have to store `userId` of our db since we can get `id` of firebase by using property of `firebase currentUser`.